### ISSUE TITLE
Portlets removed when OSGi re-inits.

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18052]
If you have an OSGi plugin that adds a portlet to dotCMS, these portlets disappear when OSGi is restarted.  What happens is that when the osgi framework restarts or when you redeploy a plugin any osgi portlet is also removed from all menu. You can see that it is being forcibly removed by `GenericBundleActivator.unregisterServices()` which is called if you restart the osgi framework and/or you redeploy the plugin. Here is where it is being undeployed

https://github.com/dotCMS/core/blob/68a3c9169f21015f18cfc43f7bf1acefcf87946c/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java#L151 

It should not be implemented like this.   We should leave the reference in the `cms_layouts_portlets` table so that when you redeploy a plugin, restart the framework or maybe even restart dotCMS you do not need to add the portlet to the menu again.  

The reference should only be removed if the user actively "undeploys" the portlet.

**Comment by fabrizzio-dotCMS** [comment]
I wasn't able to reproduce this case.
First I installed this plugin:
https://github.com/dotCMS/plugin-seeds/tree/master/OSGi/com.dotcms.portlet 
Had the plugin added to the content tab. Via  Roles and Tools > Tools
Then I tried restarting the OSGI framework and also installing an additional plugin.
The installed portlets remain installed as expected. 

Also had a breakpoint on `PortletFactoryImp.deletePortlet` And I was able to verify this only gets called when the plugin itself is removed explicitly.

I believe this behavior got fixed already on 
https://github.com/dotCMS/core/issues/17127
